modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2177,6 +2177,15 @@ public CodegenProperty fromProperty(String name, Schema p) {
             //    property.baseType = getSimpleRef(p.get$ref());
             //}
             // --END of revision
+
+            // set is enum to true when a model is enum and type is non-primitive.
+            if (StringUtils.isNotBlank(p.get$ref()) && globalSchemas != null) {
+               Schema ref = globalSchemas.get(ModelUtils.getSimpleRef(p.get$ref()));
```

Comment:
What about using `ModelUtils.unaliasSchema(globalSchemas, p)` instead?

Reply:
I have analyzed the same code: `ModelUtils.unaliasSchema(globalSchemas, p)` is returning the reference-schema to indicate that a outer-enum must be created (which is correct). 

You could use `ModelUtils.getReferencedSchema(this.openAPI, p)` to get the referenced schema regardless of any other consideration.